test/data/kms-aws/encrypt-response-partial.txt

```diff
@@ -0,0 +1,3 @@
+HTTP/1.1 200 OK
+x-amzn-RequestId: deeb35e5-4ecb-4bf1-9af5-84a54ff0af0e
+Content-Type: appli
```

Comment:
```suggestion
HTTP/1.1 200 OK
x-amzn-RequestId: deeb35e5-4ecb-4bf1-9af5-84a54ff0af0e
Content-Type: application/x-amz-json-1.1
Content-Length: 446
Connection: close

{"KeyId": "arn:aws:k
```

Suggest including all headers and part of body. Otherwise, this sequence appears to succeed:
```c
ASSERT_OK(mongocrypt_kms_feed(kms, TEST_FILE("./test/data/kms-aws/encrypt-response-partial.txt")), kms);
ASSERT_OK(mongocrypt_kms_feed(kms, TEST_FILE("./test/data/kms-aws/encrypt-response.txt")), kms);
```

I expect this is due to the extra headers being ignored. With this suggestion, the second call fails.